### PR TITLE
feat(i18n): domain routing strategy

### PR DESCRIPTION
**Pull request by ematipico**
#### Description (required)

This PR adds documentation for domain support in the i18n routing.

RFC chapter: https://github.com/withastro/roadmap/blob/i18n-routing/proposals/0041-i18n-routing.md#domain-support

 #### For Astro version: `4.0`. See astro PR [#9143](https://github.com/withastro/astro/pull/9143).

### CODE REVIEW

**Review by dreyfus92 — APPROVED** [review]
LGTM ✅

**Comment by sarah11918** [comment]
No rush on this now, but I've giving this a once-over by editing the file, which allowed me to also do some editing in other parts of the page to update for this new option. So, we can work on this from this point now!

File: src/content/docs/en/guides/internationalization.mdx [hunk 1]
```diff
@@ -159,6 +159,65 @@ Set this option when all routes will have their `/locale/` prefix in their URL a
 
 - URLs without a locale prefix, (e.g. `example.com/blog/`) will return a 404 (not found) status code.
 
+#### `'domain'`
+
+Use this strategy when you need to support multiple domains (e.g. `fr.example.com/blog`, `example.es/blog`). 
+
+This routing strategy is available only for pages that are [rendered on-demand](/en/core-concepts/rendering-modes/#on-demand-rendered).
+
+You must also specify the URL domain pattern to use in a `domains` configuration object. For any `locales` not configured in `domains`, the fallback URLs created will follow [`prefix-other-locales`](#prefix-other-locales), and create a `/[locale]/` path for all non-default languages.
```
Comment (sarah11918) [inline src/content/docs/en/guides/internationalization.mdx:168]:
```suggestion
You must also specify the URL domain pattern to use in a `domains` configuration object. For any `locales` not configured in `domains`, a `/[locale]/` path for all non-default languages will be created.
```

**Comment by sarah11918** [comment]
No hard feelings! Just closing this one until we can actually get back to it with the updated `routing` strategy in place! Feel free to reopen, or if it turns out things work a little differently, then we can start with a brand new PR!

**Pull request closed without merging**